internal/pkg/manifest/workload_ecs.go

```diff
@@ -330,6 +330,56 @@ func (cfg *SidecarConfig) ImageURI() (string, bool) {
 	return "", false
 }
 
+// BuildConfig populates a docker.BuildArguments struct from the sidecar image fields available in the manifest.
+func (cfg *SidecarImageConfig) BuildConfig(rootDirectory string) *DockerBuildArgs {
```

Comment:
If you think https://github.com/aws/copilot-cli/pull/4430/files#r1101811523 sounds good, then we probably won't need to repeat these logic from the current L166.

Reply:
correct me if am dumb, Even after implementing this https://github.com/aws/copilot-cli/pull/4430/files#r1101811523

If I want to loop over each sidecar and return the `DockerBuildArgs` like this below
```
func (s *LoadBalancedWebService) SidecarBuildArgs(wsRoot string) map[string]*DockerBuildArgs {
	sc := s.LoadBalancedWebServiceConfig.Sidecars
	sidecarBuildArgs := make(map[string]*DockerBuildArgs, len(sc))
	for k, v := range sc {
		sidecarBuildArgs[k] = v.Image.Advanced.BuildConfig(wsRoot)
	}
	return sidecarBuildArgs
}
```
then I would still need to write all these functions 
```
// BuildConfig populates a docker.BuildArguments struct from the sidecar image fields available in the manifest.
func (cfg *ImageLocationOrBuild) BuildConfig(rootDirectory string) *DockerBuildArgs {
	df := cfg.dockerfile()
	ctx := cfg.context()
	dockerfile, context := dockerfileAndContext(rootDirectory, df, ctx)
	return &DockerBuildArgs{
		Dockerfile: dockerfile,
		Context:    context,
		Args:       cfg.args(),
		Target:     cfg.target(),
		CacheFrom:  cfg.cacheFrom(),
	}
}

// dockerfile returns the path to the sidecar container's Dockerfile if one is set.
// If no dockerfile is specified, returns "".
func (cfg *ImageLocationOrBuild) dockerfile() string {
	switch {
	case cfg.Build.Advanced.Dockerfile != nil:
		return aws.StringValue(cfg.Build.Advanced.Dockerfile)
	case cfg.Build.Basic != nil:
		return aws.StringValue(cfg.Build.Basic)
	default:
		return ""
	}

}
```